### ISSUE TITLE
Fallback for API < 21?

### ISSUE THREAD

**Issue by zvoncha** [vikeri/react-native-background-job#14]
Hey, is there any chance to implement a fallback for API < 21, I belive it is AlarmManager? https://developer.android.com/reference/android/app/AlarmManager.html

**Comment by vikeri** [comment]
In our app we are only using +21 so I can't really prioritize it a the moment. But I would happily accept a pull request.

**Comment by zvoncha** [comment]
Thank you for an answer. I will certainly in incoming weeks try to implement this feature.

**Comment by vikeri** [comment]
Ok great!

**Comment by mrbarde** [comment]
Any luck?

**Comment by zvoncha** [comment]
@mrbarde Currenlty working on another project not related to this issue, I will roll up my sleeves and start working on this as soon as I finish with current project.

**Comment by jcarbo** [comment]
@zvoncha @mrbarde @vikeri - I opened a PR that implements support for this (see #18). On Android API < 21 it will still compile but won't be functional (exactly like it is on iOS).

I'm using it in my app right now like this:
```
yarn add react-native-background-job@https://github.com/jcarbo/react-native-background-job.git#feature/min-sdk-version
```

**Comment by vikeri** [comment]
@jcarbo Great, I'll look at your PR tomorrow. What I think zvoncha was referencing though was an implementation that not only compiles below API 21 but that actually works for those versions as well.

**Comment by zvoncha** [comment]
@jcarbo Thanks for this, however I need not only to compile, but to actually work on Android < 5.0 as @vikeri said. I started today again to work on this issue after longer delay, hopefully I will have some solution soon.

**Comment by vikeri** [comment]
@zvoncha Great! Beware that I made a major change in order to support "always on" background jobs. So make sure that you clone the latest version 😄

**Comment by zvoncha** [comment]
Thank @vikeri I will keep my mind on that :)

**Comment by toteto** [comment]
I just made a fork (https://github.com/toteto/react-native-background-job) that will lower the minSdk to 16.

It will be using https://github.com/firebase/firebase-jobdispatcher-android instead of the platform Job Scheduler.

Should be done soon.

**Comment by atlanteh** [comment]
@toteto any update?

**Comment by toteto** [comment]
@atlanteh Started few days with the migration, some polishing and testing left to do.

**Comment by toteto** [comment]
@atlanteh @zvoncha @yabeow
Create pull request (https://github.com/vikeri/react-native-background-job/pull/38) that fixes this and some of the other issues.

I would like to ask if you could add my fork as a dependency to your project and test if everything is working correctly. Post your findings in the pull request.

**Comment by vikeri** [comment]
Solved by #38
